packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/SubscriptionSetImpl.kt

```diff
@@ -84,7 +84,8 @@ internal class SubscriptionSetImpl<T : BaseRealm>(
         val channel = Channel<Any>(1)
         try {
             val result: Any = withTimeout(timeout) {
-                withContext((realm.configuration as SyncConfigurationImpl).notificationDispatcher) {
+                // TODO Assuming this is always a RealmImpl is probably dangerous
```

Comment:
How can we make it less dangerous?

Reply:
This should be safe until we introduce a public DynamicRealm as we only expose subscriptions through the Realm class. I expanded this comment a bit to make it more clear.